spec/spec_helper.rb

```diff
@@ -128,6 +128,11 @@ def client_assertion_with_nonexistent_client_id_path
     ActionMailer::Base.deliveries.clear
   end
 
+  # Clear Warden environment
+  config.after(:each) do
```

Comment:
Do we need a `config.after(:each)` for each one?

Also, if this fixes random rspec fails, then you should create its own small PR for it, which then probably could be merged faster than this big one.

Personally I think that also the `pry-rescue`-change should have its own PR, don't add unrelated stuff to a PR that is already big enough. If you need the changes here, you can rebase this PR on top of the other PRs.